### PR TITLE
Reduce hardcoding of NF parameters (https://github.com/nephio-project…

### PR DESCRIPTION
**Pull request by VamshiKrishnaNemalikonda**
Reduce hardcoding of NF parameters (https://github.com/nephio-project/nephio/issues/340)

### CODE REVIEW

File: nf_deployments/v1alpha1/nf_deployment_types.go [hunk 1]
```diff
@@ -34,153 +57,176 @@ type NFDeploymentSpec struct {
...
+type ObjectReference struct {
+	// APIVersion of the target resources
+	APIVersion string `yaml:"apiVersion,omitempty" json:"apiVersion,omitempty"`
+
+	// Kind of the target resources
+	Kind string `yaml:"kind,omitempty" json:"kind,omitempty"`
+
+	// Name of the target resource
+	// +optional
+	Name *string `yaml:"name" json:"name"`
```
Comment (josephthaliath) [inline nf_deployments/v1alpha1/nf_deployment_types.go:72]: Is there a reason for making it *string instead of string

Comment (VamshiKrishnaNemalikonda) [reply nf_deployments/v1alpha1/nf_deployment_types.go:72]: No I don't any particular reason as of now.

Comment (henderiw) [reply nf_deployments/v1alpha1/nf_deployment_types.go:72]: The rule we follow in the api is the reverse of what we have done here.

omit empty -> means it is optional and we make it a pointer -> so *string
if mandatory it should be a string.

I believe for us all of these parameters are mandatory

File: nf_deployments/v1alpha1/nf_deployment_types.go [hunk 2]
```diff
@@ -34,153 +57,176 @@ type NFDeploymentSpec struct {
 	NetworkInstances []NetworkInstance `json:"networkInstances,omitempty" yaml:"networkInstances,omitempty"`
 	// configRef defines addiitonal configuration references the nf depends upon
 	// +optional
-	ConfigRefs []corev1.ObjectReference `json:"configRefs,omitempty" yaml:"configRefs,omitempty"`
+	ParametersRefs []ObjectReference `json:"parametersRefs,omitempty" yaml:"parametersRefs,omitempty"`
+}
+
+type ObjectReference struct {
+	// APIVersion of the target resources
+	APIVersion string `yaml:"apiVersion,omitempty" json:"apiVersion,omitempty"`
```
Comment (henderiw) [inline nf_deployments/v1alpha1/nf_deployment_types.go:65]: apiversion, kind and name should be mandatory in my view

File: nf_deployments/v1alpha1/nf_deployment_types.go (hunk 1 above)
Comment (josephthaliath) [reply nf_deployments/v1alpha1/nf_deployment_types.go:72]: Thanks Wim for the clarification. I also think these should be mandatory. The kind will always be Config right? the embedded struct inside spec.config will be a type that is known within the controller. But if there are multiple Config and different embedded structures, how can we map it to right structures. Should the embedded struct always have a key 'kind' which will be used for this purpose. Or do we add this info in ParametersRef (like embedded kind)

File: nf_deployments/v1alpha1/nf_deployment_types.go (hunk 2 above)
Comment (VamshiKrishnaNemalikonda) [reply nf_deployments/v1alpha1/nf_deployment_types.go]: I've modified as suggested @henderiw @josephthaliath

File: nf_deployments/v1alpha1/nf_deployment_types.go (hunk 1 above)
Comment (henderiw) [reply nf_deployments/v1alpha1/nf_deployment_types.go]: No this kind is the kind of the KRM resource we reference. So e.g. if we define 3gpp.org/v1alpha1.XXX name=foo

the kind will be XXX apiVersion: 3gpp.org/v1alpha1 and name foo

**Comment by henderiw** [comment]
prow /retest

File: config/crd/bases/workload.nephio.org_nfdeployments.yaml [hunk 3]
```diff
@@ -314,8 +251,28 @@ spec:
                   - name
                   type: object
                 type: array
+              parametersRefs:
+                description: configRef defines addiitonal configuration references
```
Comment (efiacor) [inline config/crd/bases/workload.nephio.org_nfdeployments.yaml:255]: "parametersRef defines additional"

Comment (henderiw) [reply config/crd/bases/workload.nephio.org_nfdeployments.yaml:255]: can you approve this PR https://github.com/nephio-project/api/pull/53
it addresses your comment

**Pull request merged**